### PR TITLE
Clean up of event filter expression parser (#324)

### PR DESCRIPTION
**Pull request by joachimmetz**
Clean up of event filter expression parser

* removed SelfFeederMixIn
* merged Lexer and SearchParser classes
* changes to docstrings

### CODE REVIEW

**Comment by joachimmetz** [comment]
@Onager PTAL, if you have comments on the docstrings, please add a suggestion, some of them are educated guesses. The idea is to clean up this file more and improve the docstrings in the process.

File: plaso/lib/lexer.py [hunk 1]
```diff
@@ -12,26 +12,22 @@
...
 
     Args:
-
-      state_regex: If this regular expression matches the current state this
-                   rule is considered.
-      regex: A regular expression to try and match from the current point.
-      actions: A command separated list of method names in the Lexer to call.
-      next_state: The next state we transition to if this Token matches.
-      flags: re flags.
+      state_regex (str): regular expression that is considered when the current
```
Comment (Onager) [inline plaso/lib/lexer.py]: This is an re, not a string. The type seems to in be re._pattern_type, but the exact name looks different on Python 2 and 3.

File: plaso/lib/lexer.py [hunk 2]
```diff
@@ -12,26 +12,22 @@
...
+          state matches this rule.
+      regex (str): regular expression to try and match from the current point.
```
Comment (Onager) [inline plaso/lib/lexer.py]: This is an re too, not a string.

File: plaso/lib/lexer.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/lib/lexer.py]: this will be removed in a later change

File: plaso/lib/lexer.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/lib/lexer.py]: this will be made protected in a later change

**Pull request merged**